src/irrlicht_changes/CGUITTFont.cpp

```diff
@@ -632,8 +634,10 @@ void CGUITTFont::draw(const EnrichedString &text, const core::rect<s32>& positio
 			CGUITTGlyphPage* const page = Glyph_Pages[glyph.glyph_page];
 			page->render_positions.push_back(core::position2di(offset.X + offx, offset.Y + offy));
 			page->render_source_rects.push_back(glyph.source_rect);
-			if (iter.getPos() < colors.size())
-				page->render_colors.push_back(colors[iter.getPos()]);
+
+			// static_cast<unsigned long>(iter - utext.begin()) is a nasty hack to get the position of iter in the wtext string.
```

Comment:
I don't think iterator arithmetic is hack, rather the intended way?
also you use this value multiple times so I suggest declaring a variable